reference-files/plugin.sketchplugin/Contents/Sketch/groups.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable */
+
+const {
+  Document,
+  Artboard,
+  Shape,
+  Rectangle,
+  Style,
+  Text,
+  Group,
+} = require('sketch/dom')
+
+function main(ctx) {
+  // Create a new document, and get a reference to its selected page
+  const doc = new Document()
+  const page = doc.selectedPage
+
+  // Create an artboard
+  const artboard = new Artboard({
+    name: 'MyArtboard',
+    parent: page,
+    frame: new Rectangle(0, 0, 200, 200),
+  })
+
+  // Nest a group within the artboard containing two additional layers
+  var group = new Group({
+    name: 'MyGroup',
+    parent: artboard,
+    layers: [
+      new Text({
```

Comment:
not that it's an issue but you don't have to create the object here, you could specify the json representation directly:
```
var group = new Group({
     name: 'MyGroup',
     parent: artboard,
     layers: [
      {
        type: 'Text',
        text: 'Hello World',
         style: {
           borders: [],
           textColor: '#f00',
           fontSize: 28,
           fontFamily: 'Arial',
           fontWeight: 10,
         },
       },
       {
         type: 'Shape',
         name: 'MyShape',
         frame: new Rectangle(0, 50, 50, 50),
         style: {
           opacity: 0.5,
           borders: [],
           fills: [{ color: '#0f0', fillType: Style.FillType.Color }],
         },
       }
})
```

You could even specify all of that in the Artboard constructor

Reply:
Ah interesting. Good to know!

Reply:
same for `rectangle` actually: you don't need the constructor, you can pass an json object directly